staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go

```diff
@@ -83,14 +85,18 @@ type Options struct {
 // BuildSwagger builds swagger for the given crd in the given version
 func BuildSwagger(crd *apiextensions.CustomResourceDefinition, version string, opts Options) (*spec.Swagger, error) {
 	var schema *structuralschema.Structural
-	s, err := apiextensions.GetSchemaForVersion(crd, version)
+	s, err := apiextensionshelpers.GetSchemaForVersion(crd, version)
 	if err != nil {
 		return nil, err
 	}
 
 	if s != nil && s.OpenAPIV3Schema != nil {
-		if !validation.SchemaHasInvalidTypes(s.OpenAPIV3Schema) {
-			if ss, err := structuralschema.NewStructural(s.OpenAPIV3Schema); err == nil {
+		internalCRDSchema := &apiextensionsinternal.CustomResourceValidation{}
+		if err := apiextensions.Convert_v1_CustomResourceValidation_To_apiextensions_CustomResourceValidation(s, internalCRDSchema, nil); err != nil {
```

Comment:
highlight explicit conversion here: the builder function does another round of validation to defeat the invalid openapi property type. all validation works on internal types, so we need conversion here